src/main/java/edu/cornell/kfs/sys/batch/JAXBXmlBatchInputFileTypeBase.java

```diff
@@ -0,0 +1,113 @@
+import java.util.Date;
+
+import javax.xml.bind.JAXBException;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.batch.BatchInputFileTypeBase;
+import org.kuali.kfs.sys.exception.ParseException;
+import org.kuali.rice.core.api.datetime.DateTimeService;
+
+import edu.cornell.kfs.sys.service.CUMarshalService;
+
+/**
+ * Alternative to XmlBatchInputFileTypeBase that parses XML POJOs using JAXB instead.
+ * 
+ * Similarly to FlatFileParserBase, this implementation has properties to configure
+ * the File Type Identifier and Title Key, and also constructs file names using
+ * the same pattern as that class (and has a property for the file name prefix).
+ */
+public class JAXBXmlBatchInputFileTypeBase extends BatchInputFileTypeBase {
+
+    private static final String FILE_NAME_PART_DELIMITER = "_";
+
+    protected DateTimeService dateTimeService;
+    protected CUMarshalService cuMarshalService;
+    protected String fileTypeIdentifier;
+    protected String titleKey;
+    protected String fileNamePrefix;
+    protected Class<?> pojoClass;
+
+    @Override
+    public String getFileName(String principalName, Object parsedFileContents, String fileUserIdentifier) {
+        Date currentDate = dateTimeService.getCurrentDate();
+        String fileName = new StringBuilder()
+                .append(fileNamePrefix)
+                .append(principalName)
+                .append(buildFileNameFragmentFromFileUserIdentifier(fileUserIdentifier))
+                .append(FILE_NAME_PART_DELIMITER).append(dateTimeService.toDateTimeStringForFilename(currentDate))
+                .toString();
+        
+        return StringUtils.remove(fileName, KFSConstants.BLANK_SPACE);
+    }
+
+    protected String buildFileNameFragmentFromFileUserIdentifier(String fileUserIdentifier) {
+        return StringUtils.isNotBlank(fileUserIdentifier)
+                ? FILE_NAME_PART_DELIMITER + fileUserIdentifier
+                : StringUtils.EMPTY;
+    }
+
+    @Override
+    public String getFileTypeIdentifer() {
+        return fileTypeIdentifier;
+    }
+
+    @Override
+    public Object parse(byte[] fileByteContent) throws ParseException {
+        try {
+            String fileStringContent = new String(fileByteContent, StandardCharsets.UTF_8);
+            return cuMarshalService.unmarshalString(fileStringContent, pojoClass);
+        } catch (JAXBException e) {
+            throw new ParseException("Error attempting to unmarshal POJO from XML", e);
+        }
+    }
+
+    @Override
+    public void process(String fileName, Object parsedFileContents) {
+        // Do nothing by default.
+    }
+
+    @Override
+    public boolean validate(Object parsedFileContents) {
+        return true;
+    }
+
+    @Override
+    public String getAuthorPrincipalName(File file) {
+        return StringUtils.substringBetween(file.getName(), fileNamePrefix, FILE_NAME_PART_DELIMITER);

```

Comment:
should this include some error handling?  What do we want to do if the principle name can't be parsed from the file name?

Reply:
I had based this method implementation off of the one from FlatFileParserBase (which also extends from BatchInputFileTypeBase). I think KFS's other implementations of this method simply return null if the principal name couldn't be found in the file. The StringUtils.substringBetween() method should already be doing much of the error handling here; if you think it's worth adding a null check on the File object as well, please let me know.

Reply:
might as well stay consistent with other KFS documents, I just don't want to fail to create a document because of a file naming problem